app/services/chatbot_integration.py

```
@@ -0,0 +1,140 @@
+            .join(User, Voice.user_id == User.user_id)
+            .options(joinedload(Voice.questions))
+            .filter(Voice.voice_id == voice_id)
+            .first()
+        )
+
+    except SQLAlchemyError as e:
+        # DB 연결 끊김, 쿼리 구문 오류 등 SQLAlchemy 관련 심각한 오류 발생
+        logger.error(f"[ChatbotIntegration] Database query failed for voice_id={voice_id}: {e}", exc_info=True)
+        raise DatabaseException(
+            message="Failed to retrieve data due to an internal database error."
+        )
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)